src/server/worker/datum_iterator_test.go

```diff
@@ -0,0 +1,78 @@
+package worker
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/pachyderm/pachyderm/src/client"
+	"github.com/pachyderm/pachyderm/src/client/pkg/require"
+	tu "github.com/pachyderm/pachyderm/src/server/pkg/testutil"
+)
+
+func TestDatumIterators(t *testing.T) {
```

Comment:
I think it would be a good idea to expand on this test a bit more. Right now I think it just checks for duplicates and that the number of datums read match the length. It seems to make more sense to me that we should have a hard coded, expected result (list of datums) for each of the types so that we can confirm the order and the actual datums produced to validate that the iterators work as expected.